docs/plugins/plugin-script.asciidoc

```diff
@@ -0,0 +1,229 @@
+-----------------------------------
+bin/plugin --url file:///path/to/plugin.zip --install my_plugin
+-----------------------------------
+
+.Installation process
+*************************************
+
+The plugin installer tries to download a Zip file from a custom URL (if
+specified), from `download.elastic.co`, Maven Central, Sonatype, or GitHub.
+If successful, it unzips the plugin into the `plugins/` directory, usually
+located under `$ES_HOME`.
+
+If a plugin contains a `bin/` or `config/` directory, their contents are moved
+to the `$ES_HOME/bin` and `$ES_HOME/config` directories respectively.
+
+If any other non-JAR files are present and the plugin doesn't already contain
+a `_site/` directory, that directory is created and all non JAR files are moved
+into it.
+
+*************************************
+
+
+[[listing-removing]]
+=== Listing and Removing Installed Plugins
+
+[float]
+==== Listing plugins
+
+A list of the currently loaded plugins can be retrieved with the `--list` option:
+
+[source,shell]
+-----------------------------------
+bin/plugin --list
+-----------------------------------
+
+Alternatively, use the {ref}/cluster-nodes-info.html[node-info API] to find
+out which plugins are installed on each node in the cluster
+
+[float]
+==== Removing plugins
+
+Plugins can be removed manually, by deleting the appropriate directory under
+`plugins/`, or using the public script:
+
+[source,shell]
+-----------------------------------
+plugin --remove [pluginname]
+-----------------------------------
+
+=== Other command line parameters
+
+The `plugin` scripts supports a number of other command line parameters:
+
+[float]
+==== Silent/Verbose mode
+
+The `--verbose` parameter outputs more debug information, while the `--silent`
+parameter turns off all output.  The script may return the following exit
+codes:
+
+[horizontal]
+`0`:: everything was OK
+`64`:: unknown command or incorrect option parameter
+`74`:: IO error
+`70`:: any other error
+
+[float]
+==== Timeout settings
+
+By default, the `plugin` script will wait indefinitely when downloading before
+failing. The timeout parameter can be used to explicitly specify how long it
+waits. Here is some examples of setting it to different values:
+
+[source,shell]
+-----------------------------------
+# Wait for 30 seconds before failing
+bin/plugin --install mobz/elasticsearch-head --timeout 30s
+
+# Wait for 1 minute before failing
+bin/plugin --install mobz/elasticsearch-head --timeout 1m
+
+# Wait forever (default)
+bin/plugin --install mobz/elasticsearch-head --timeout 0
+-----------------------------------
+
+[float]
+==== Proxy settings
+
+To install a plugin via a proxy, you can pass the proxy details in with the
+Java settings `proxyHost` and `proxyPort`. On Unix based systems, these
+options can be set on the command line:
+
+[source,shell]
+-----------------------------------
+bin/plugin -DproxyHost=host_name -DproxyPort=port_number --install mobz/elasticsearch-head
```

Comment:
Unsure about this one. I don't know if -Dxxx needed to be set before or after install command:

```sh
bin/plugin install mobz/elasticsearch-head -DproxyHost=host_name -DproxyPort=port_number 
```